velox/functions/prestosql/GreatestLeast.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/Expr.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+
+namespace {
+
+template <bool>
+class ExtremeValueFunction;
+
+using LeastFunction = ExtremeValueFunction<true>;
+using GreatestFunction = ExtremeValueFunction<false>;
+
+/**
+ * This class implements two functions:
+ *
+ * greatest(value1, value2, ..., valueN) → [same as input]
+ * Returns the largest of the provided values.
+ *
+ * least(value1, value2, ..., valueN) → [same as input]
+ * Returns the smallest of the provided values.
+ **/
+template <bool isLeast>
+class ExtremeValueFunction : public exec::VectorFunction {
+ private:
+  template <typename T>
+  bool shouldOverride(const T& currentValue, const T& candidateValue) const {
+    return isLeast ? candidateValue < currentValue
+                   : candidateValue > currentValue;
+  }
+
+  template <typename T>
+  void applyTyped(
```

Comment:
I'm seeing that least(1, 2.0, 3, nan()) in Presto returns an error: Invalid argument to least(): NaN. Would you make sure Velox also returns an error?